### PR TITLE
Analyze parameters

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: analyzer/analyzer_parameters.h [hunk 1]
```diff
@@ -0,0 +1,121 @@
+#ifndef ANALYZER_PARAMETERS_H
+#define ANALYZER_PARAMETERS_H
+
+/*
+ * analyzer_parameters
+ *
+ * Generally other analyzers should cover bad configuration via
+ * parameters.  Until we have an analyzer which seems to cover a
+ * paremeter value check, 'though, it gets thrown in here
```
Comment (hamishwillee) [inline analyzer/analyzer_parameters.h]: Typo - 'though,

**Comment by hamishwillee** [comment]
I think this fails the "better output test" because you have a "Pass" that sounds like a fail. It is ambiguous/confusing.

```
TEST: Parameters
Description: Autopilots store configuration settings known as 'parameters'. This test detects configuration issues present in these parameters.
* Pass: Calibration value for the current sensor incorrect.  Power bricks vary in their measurements.  Unless calibrated, a true representation of the craft's remaining power will not be available.
```

IN the comments you have:
```
+ * Generally other analyzers should cover bad configuration via
+ * parameters.  Until we have an analyzer which seems to cover a
+ * paremeter value check, 'though, it gets thrown in here
```
Should we capture in the description that this is not exhaustive - it just catches the ones that we know about that are not covered by other tests?

**Comment by peterbarker** [comment]
@hamishwillee I've corrected some of the wording on this.

I'm not sure where that "Pass that sounds like a fail" output appears.  The emitted JSON looks correct to me.

**Comment by hamishwillee** [comment]
Me either. This looks OK to me now.

**Pull request merged**